python_modules/dagster/dagster/_core/host_representation/external.py

```diff
@@ -220,15 +222,29 @@ def _external_sensors(self) -> Dict[str, "ExternalSensor"]:
                 if asset_graph.get_auto_observe_interval_minutes(asset_key) is None:
                     continue
 
+                has_any_auto_observe_source_assets = True
+
                 if asset_key not in covered_asset_keys:
                     default_sensor_asset_keys.add(asset_key)
 
             if default_sensor_asset_keys:
+                # Use AssetSelection.all if the default sensor is the only sensor - otherwise
+                # enumerate the assets that are not already included in some other
+                # non-default sensor
+                default_sensor_asset_selection = AssetSelection.all(
+                    include_sources=has_any_auto_observe_source_assets
```

Comment:
i kept this too because it seems nice to leave out the "and observable assets" part of the string if its not relevant